tests/integration/pox/stacking/direct-stacking-good-order.spec.ts

```diff
@@ -0,0 +1,122 @@
+  broadcastStackSTX,
+} from "../helpers-direct-stacking";
+import {
+  OptionalCV,
+  SomeCV,
+  TupleCV,
+  cvToString,
+  hexToCV,
+} from "@stacks/transactions";
+
+describe("testing solo stacker increase without bug", () => {
+  let orchestrator: DevnetNetworkOrchestrator;
+  let timeline = {
+    epoch_2_0: 100,
+    epoch_2_05: 102,
+    epoch_2_1: 106,
+    pox_2_activation: 109,
+  };
+
+  beforeAll(() => {
+    orchestrator = buildDevnetNetworkOrchestrator(getNetworkIdFromEnv());
+    orchestrator.start();
+  });
+
+  afterAll(() => {
+    orchestrator.terminate();
+  });
+
+  it("using stacks-increase in the same cycle should result in increased rewards", async () => {
+    const network = new StacksTestnet({ url: orchestrator.getStacksNodeUrl() });
+
+    // Wait for Stacks genesis block
+    await orchestrator.waitForNextStacksBlock();
+    // Wait for block N+1 where N is the height of the next reward phase
+    await waitForNextRewardPhase(network, orchestrator, 1);
+
+    const blockHeight = timeline.pox_2_activation + 1;
+    const fee = 1000;
+    const cycles = 1;
+
+    // Bob stacks 30m
+    let response = await broadcastStackSTX(
+      2,
+      network,
+      30_000_000_000_010,
+      Accounts.WALLET_2,
+      blockHeight,
+      cycles,
+      fee,
+      0
+    );
+    expect(response.error).toBeUndefined();
+
+    // Bob increases by 20m
+    response = await broadcastStackIncrease(
+      network,
+      20_000_000_000_100,
+      Accounts.WALLET_2,
+      fee,
+      1
+    );
+    expect(response.error).toBeUndefined();
+
+    // let Bob's stacking confirm to enforce reward index 0
+    await waitForStacksTransaction(orchestrator, response.txid);
+
+    // Alice stacks 50m
+    response = await broadcastStackSTX(
+      2,
+      network,
+      50_000_000_000_001,
+      Accounts.WALLET_1,
+      blockHeight,
+      cycles,
+      fee,
+      0
+    );
+    expect(response.error).toBeUndefined();
+
+    await orchestrator.waitForStacksBlockIncludingTransaction(response.txid);
+    let poxInfo = await getPoxInfo(network);
+
+    // Asserts about pox info for better knowledge sharing
+    expect(poxInfo.contract_id).toBe("ST000000000000000000002AMW42H.pox-2");
+    expect(poxInfo.pox_activation_threshold_ustx).toBe(50_286_942_145_278);
+    expect(poxInfo.current_cycle.id).toBe(1);
+    expect(poxInfo.current_cycle.min_threshold_ustx).toBe(20_960_000_000_000);
+
+    // Assert that the next cycle has 100m STX locked
+    expect(poxInfo.current_cycle.stacked_ustx).toBe(0);
+    expect(poxInfo.current_cycle.is_pox_active).toBe(false);
+    expect(poxInfo.next_cycle.stacked_ustx).toBe(100_000_000_000_111);
+
+    const poxAddrInfo0 = await readRewardCyclePoxAddressList(network, 2, 0);
+    const poxAddrInfo0CV = hexToCV(poxAddrInfo0.data) as SomeCV<TupleCV>;
+    const poxAddrInfoStr0 = cvToString(hexToCV(poxAddrInfo0.data));
+    // There is no bug here because total stack was 0 when stack-increase was called.
```

Comment:
Should this say, "There is no bug here because total stack was equal to Bob's stacked amount when Bob called stack-increase."?

Reply:
I meant total stack of all stackers.

Reply:
Right, but isn't the total stacked equal to Bob's total stacked, not 0?

Reply:
I meant the total stacked amount of all stackers

Reply:
I see. Comment has been corrected.